drivers/ethernet/eth_stm32_hal.c

```diff
@@ -953,6 +951,15 @@ static int eth_initialize(const struct device *dev)
 		LOG_ERR("Could not configure ethernet pins");
 		return ret;
 	}
+	return 0;
+}
+
+static int hal_init(void)
+{
+	HAL_StatusTypeDef hal_ret = HAL_OK;
```

Comment:
super nit but, this could go last, to have reverse xmas tree declarations

Reply:
I didn't think about this, thanks! If I understand it good - this variable definition should be placed at the end of variable initialization area. See commit `#`